### PR TITLE
docs: update "upgrading node" guide

### PR DESCRIPTION
**Pull request by nornagon**
##### Description of Change
Remove some references to libcc and add some notes about the GN build files we now maintain.

I think there's more to do here, and I'll continue to add to & update this documentation as I go through the process of upgrading node in the context of the GN build.

##### Checklist
- [x] PR description included and stakeholders cc'd
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

##### Release Notes
Notes: no-notes

### CODE REVIEW

File: docs/development/upgrading-node.md [hunk 1]
```diff
@@ -2,14 +2,15 @@
 
 ## Discussion
 
-One upgrade issue is building all of Electron with a single copy
-of V8 to ensure compatibility. This is important because
-upstream Node and [libchromiumcontent](upgrading-chromium.md)
-both use their own versions of V8.
+Chromium and Node.js both depend on V8, and Electron contains
+only a single copy of V8, so it's important to ensure that version
```
Comment (jkleinsc) [inline docs/development/upgrading-node.md]: > so it's important to ensure that version 

should be 

> so it's important to ensure that the version

File: docs/development/upgrading-node.md [hunk 2]
```diff
@@ -70,49 +66,35 @@ We need to generate a patch file from each patch applied to V8.
 4. Inspect Node [repo](https://github.com/electron/node) to see what patches upstream Node
   used with their v8 after bumping its version
   - `git log --oneline "deps/v8"`
+  - The commit message "update V8 to <version>" represents replacing `deps/v8` with the named version of V8, exactly as it is in upstream. The commit message "patch V8 to <version>" isn't a complete replacement of `deps/v8`, but just applying the changes in that version.
 5. Create a checklist of the patches. This is useful for tracking your work and for
   having a quick reference of commit hashes to use in the `git diff-tree` step below.
-6. Read `patches/common/v8/README.md` to see which patchfiles came from the previous version of V8 and therefore need to be removed.
-  - Delete each patchfile referenced in `patches/common/v8/README.md`
-7. Apply all patches with the [`get-patch` script](https://github.com/electron/libchromiumcontent/blob/master/script/README.md#get-patch):
+6. Apply all patches with the [`get-patch` script](https://github.com/electron/electron/blob/master/script/README.md#get-patch):
```
Comment (alexeykuzmin) [inline docs/development/upgrading-node.md]: It shouldn't say "Apply all patches..." but rather "Save all commits as patch files...".

And maybe this whole recommendation should replaced with your solution with `git format-patch --relative=deps/v8 ...`

File: docs/development/upgrading-node.md [hunk 3]
```diff
@@ -70,49 +66,35 @@ We need to generate a patch file from each patch applied to V8.
...
 
 ## Notes
 
-- libcc and V8 are treated as a single unit
 - Node maintains its own fork of V8
   - They backport a small amount of things as needed
   - Documentation in node about how [they work with V8](https://nodejs.org/api/v8.html)
 - We update code such that we only use one copy of V8 across all of electron
-  - E.g electron, libcc, and node
+  - E.g electron, chromium, and node
```
Comment (alexeykuzmin) [inline docs/development/upgrading-node.md]: capital letter for "chromium"?

File: docs/development/upgrading-node.md (hunk 1 above)
Comment (ckerr) [reply docs/development/upgrading-node.md]: 'That version' refers back to the single copy of v8, so I don't think this is incorrect.

Could be clearer to make that reference more explicit, e.g. "so it's important to ensure the single copy of v8 is compatible..."

File: docs/development/upgrading-node.md [hunk 4]
```diff
@@ -2,14 +2,15 @@
...
 ## Discussion
 
-One upgrade issue is building all of Electron with a single copy
-of V8 to ensure compatibility. This is important because
-upstream Node and [libchromiumcontent](upgrading-chromium.md)
-both use their own versions of V8.
+Chromium and Node.js both depend on V8, and Electron contains
+only a single copy of V8, so it's important to ensure that version
+of V8 is compatible with both Node.js and the version of Chromium
+in a build.
```
Comment (ckerr) [inline docs/development/upgrading-node.md]: Slightly better: "with build's versions of Node.js and Chromium"

File: docs/development/upgrading-node.md (hunk 3 above)
Comment (ckerr) [reply docs/development/upgrading-node.md]: Not new to this PR, but `Electron` and `Node.js` as well

File: docs/development/upgrading-node.md [hunk 5]
```diff
@@ -70,49 +66,35 @@ We need to generate a patch file from each patch applied to V8.
...
-  of V8 that’s closest to the version in libcc that we’re using.
+  of V8 that’s closest to the version in Chromium that we’re using.
   - We sometimes have to wait for the next periodic Node release
-   because it will sync more closely with the version of V8 in the new libcc
- - Electron keeps all its patches in libcc because it’s simpler than
+   because it will sync more closely with the version of V8 in the new Chromium
+ - Electron keeps all its patches in the repo because it’s simpler than
    maintaining different repos for patches for each upstream project.
-   - Crashpad, node, libcc, etc. patches are all kept in the same place
+   - Crashpad, node, chromium, skia etc. patches are all kept in the same place
```
Comment (ckerr) [inline docs/development/upgrading-node.md]: Node.js, Chromium, Skia here too

File: docs/development/upgrading-node.md [hunk 6]
```diff
@@ -70,49 +66,35 @@ We need to generate a patch file from each patch applied to V8.
...
  - Building node:
-   - There’s a chance we need to change our build configuration
-   to match the build flags that node wants in `node/common.gypi`
+   - We maintain our own GN build files for node.js to make it easier to ensure
+     that node, chromium and electron are all built with the same compiler flags.
```
Comment (ckerr) [inline docs/development/upgrading-node.md]: Node.js, Chromium, and Electron -- or maybe just "everything is built with the same compiler flags" :smile:

File: docs/development/upgrading-node.md [hunk 7]
```diff
@@ -70,49 +66,35 @@ We need to generate a patch file from each patch applied to V8.
...
- - Electron keeps all its patches in libcc because it’s simpler than
+   because it will sync more closely with the version of V8 in the new Chromium
+ - Electron keeps all its patches in the repo because it’s simpler than
    maintaining different repos for patches for each upstream project.
-   - Crashpad, node, libcc, etc. patches are all kept in the same place
+   - Crashpad, node, chromium, skia etc. patches are all kept in the same place
  - Building node:
-   - There’s a chance we need to change our build configuration
-   to match the build flags that node wants in `node/common.gypi`
+   - We maintain our own GN build files for node.js to make it easier to ensure
```
Comment (ckerr) [inline docs/development/upgrading-node.md]: Node.js

File: docs/development/upgrading-node.md [hunk 8]
```diff
@@ -32,9 +33,8 @@ So in short, the primary steps are:
 
 1. Update Electron's Node fork to the desired version
 2. Backport Node's V8 patches to our copy of V8
-3. Update Electron to use new version of Node
-  * Update submodules
-  * Update Node.js build configuration
+3. Update the GN build files, porting changes from node's GYP files
```
[the remark this answers is not in the document]
Comment (nornagon) [reply docs/development/upgrading-node.md]: I'm planning to detail the GN build files we maintain in a later update.

File: docs/development/upgrading-node.md (hunk 2 above)
Comment (nornagon) [reply docs/development/upgrading-node.md]: Good idea, I've updated this to use the format-patch command.

**Comment by nornagon** [comment]
docs-only change & this doesn't touch any APIs, so merging even though archaeologist broke

**Pull request merged**